src/control/cmd/daos/health.go

```diff
@@ -0,0 +1,147 @@
+//
+// (C) Copyright 2024 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/daos-stack/daos/src/control/build"
+	"github.com/daos-stack/daos/src/control/cmd/daos/pretty"
+	"github.com/daos-stack/daos/src/control/lib/daos"
+)
+
+type healthCmds struct {
+	Check healthCheckCmd `command:"check" description:"Perform DAOS system health checks"`
+}
+
+type healthCheckCmd struct {
+	daosCmd
+	Verbose bool `short:"v" long:"verbose" description:"Display more detailed system health information"`
+}
+
+func (cmd *healthCheckCmd) Execute([]string) error {
+	// TODO (DAOS-10028): Move this logic into the daos package once the API is available.
+	systemHealth := &daos.SystemHealthInfo{
+		ComponentBuildInfo: map[build.Component]daos.ComponentBuild{
+			build.ComponentClient: {
+				Version:   build.DaosVersion,
+				BuildInfo: build.BuildInfo,
+			},
+		},
+		Pools:      make(map[uuid.UUID]*daos.PoolInfo),
+		Containers: make(map[uuid.UUID][]*daos.ContainerInfo),
+	}
+	srvBuild, err := srvBuildInfo()
+	if err != nil {
+		return err
+	}
+	systemHealth.ComponentBuildInfo[build.ComponentServer] = daos.ComponentBuild{
+		Version:   srvBuild.Version,
+		BuildInfo: srvBuild.BuildInfo,
+	}
+
+	sysInfo, err := cmd.apiProvider.GetSystemInfo()
+	if err != nil {
+		cmd.Errorf("failed to query system information: %v", err)
+	}
+	systemHealth.SystemInfo = sysInfo
+
+	cmd.Infof("Checking DAOS system: %s (Access Points: %v)", systemHealth.SystemInfo.Name, systemHealth.SystemInfo.AccessPoints())
+
+	pools, err := getPoolList(cmd.Logger, cmd.SysName, true)
+	if err != nil {
+		return err
+	}
+
+	for _, pool := range pools {
+		systemHealth.Pools[pool.UUID] = pool
+
+		poolHdl, _, err := poolConnect(pool.UUID.String(), cmd.SysName, daos.PoolConnectFlagReadOnly, false)
+		if err != nil {
+			cmd.Errorf("failed to connect to pool %s: %v", pool.Label, err)
+			continue
+		}
+		defer func() {
+			if err := poolDisconnectAPI(poolHdl); err != nil {
+				cmd.Errorf("failed to disconnect from pool %s: %v", pool.Label, err)
+			}
+		}()
+
+		queryMask := daos.MustNewPoolQueryMask(daos.PoolQueryOptionEnabledEngines)
+		tpi, err := queryPool(poolHdl, queryMask)
+		if err != nil {
+			cmd.Errorf("failed to query pool %s: %v", pool.Label, err)
+			continue
+		}
+		pool.EnabledRanks = tpi.EnabledRanks
+
+		if pool.DisabledTargets > 0 {
+			queryMask.ClearAll()
+			queryMask.SetOptions(daos.PoolQueryOptionDisabledEngines)
+			tpi, err = queryPool(poolHdl, queryMask)
+			if err != nil {
+				cmd.Errorf("failed to query pool %s: %v", pool.Label, err)
+				continue
+			}
+			pool.DisabledRanks = tpi.DisabledRanks
+		}
+
+		poolConts, err := listContainers(poolHdl)
```

Comment:
I know pool list includes only pools that the user can access. Is the same true of container list?

Reply:
Hmm... Not sure. If that's not the case, it probably wouldn't be a bad idea to make it work that way, though.

Reply:
Looking at the implementation of [ds_pool_list_cont_handler()](https://github.com/daos-stack/daos/blob/master/src/pool/srv_pool.c#L4373), I don't see any evidence that the user's credential is checked against the container ACLs before returning the list, but maybe it's happening somewhere else?

I think @kccain added this feature. There is also the [ds_pool_filter_cont_handler()](https://github.com/daos-stack/daos/blob/master/src/pool/srv_pool.c#L4519) ... Maybe that can use the ACL check as a filtering criterion?